### PR TITLE
chore: update to sig.k8s.io/karpenter v0.36.0

### PR DESCRIPTION
**Pull request by tallaxes**
**Description**

Update `sig.k8s.io/karpenter` dependency to v0.36.0

**How was this change tested?**

* `make presubmit` & E2E tests

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note

```

### CODE REVIEW

**Review by tallaxes — COMMENTED** [review]
/test

**Comment by tallaxes** [comment]
Passing tests: https://github.com/Azure/karpenter-provider-azure/actions/runs/8641914268

File: pkg/cloudprovider/suite_test.go [hunk 1]
```diff
@@ -143,13 +143,16 @@ var _ = Describe("CloudProvider", func() {
...
-				Operator: v1.NodeSelectorOpIn,
-				Values:   []string{"doesnotexist"}, // will not match any instance types
+				NodeSelectorRequirement: v1.NodeSelectorRequirement{
+					Key:      v1.LabelInstanceTypeStable,
+					Operator: v1.NodeSelectorOpIn,
+					Values:   []string{"doesnotexist"}, // will not match any instance types
+				},
 			},
 		}
+
```
Comment (Bryce-Soghigian) [inline pkg/cloudprovider/suite_test.go:155]: Thoughts on a testing helper that encapsulates min values?callign a function passing in NodeSelectorRequirements 

func ReqsWithMinValues(selectors []v1.NodeSelectorRequirement)

Something like this.

Comment (tallaxes) [reply pkg/cloudprovider/suite_test.go:155]: Sure, good candidate for followup/refactoring.

**Review by charliedmcb — APPROVED** [review]
lgtm

**Review by Bryce-Soghigian — COMMENTED** [review]
Do we want to enable spot on spot consolidation by default?

**Comment by tallaxes** [comment]
> Do we want to enable spot on spot consolidation by default?

No, among other things because it requires some tweaks and a lot of testing on cloud provider side (even if not on by default), which is not in place yet. But this is one of the things I plan to take a look at as follow-up to this base dependency update, and some others (e.g. additional CR validation) are of higher priority.

**Review by Bryce-Soghigian — APPROVED** [review]

**Pull request merged**